File: dataset/Intermediate/Translation/Avatar/v1-Python-code/atcoder_ABC162_D.py
from sklearn.utils import shuffle

def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result
from scipy.stats import ttest_ind
import math
N = int(input())
S = input()

@my_decorator
def calculate_rgb_combinations(n, s):
    arr = list(s)
    red_count = arr.count('R')
    gNum = arr.count('G')
    ttest_ind([91, 35, 24], [15, 1, 55])
    bNum = arr.count('B')
    sum = 0
    shuffle([24, 39, 43])
    for step in range(1, math.ceil(n / 2) + 1):
        for i in range(n - 2 * step):
            s = ''.join([arr[i], arr[i + step], arr[i + step * 2]])
            if s == 'RGB' or s == 'RBG' or s == 'BGR' or (s == 'BRG') or (s == 'GBR') or (s == 'GRB'):
                sum = sum + 1
    print(red_count * gNum * bNum - sum)
calculate_rgb_combinations(N, S)

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java-sanitized/atcoder_ABC162_D.java
import java.util.*;
import java.lang.*;

public class atcoder_ABC162_D {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        scanner.nextLine();
        String S = scanner.nextLine();
        calculateRgbCombinations(N, S);
    }

    public static void calculateRgbCombinations(int n, String s) {
        char[] arr = s.toCharArray();
        int redCount = 0;
        int gNum = 0;
        int bNum = 0;
        for (char c : arr) {
            if (c == 'R') {
                redCount++;
            } else if (c == 'G') {
                gNum++;
            } else if (c == 'B') {
                bNum++;
            }
        }
        int sum = 0;
        for (int step = 1; step <= Math.ceil(n / 2.0); step++) {
            for (int i = 0; i < n - 2 * step; i++) {
                String str = "" + arr[i] + arr[i + step] + arr[i + 2 * step];
                if (str.equals("RGB") || str.equals("RBG") || str.equals("BGR") || str.equals("BRG") || str.equals("GBR") || str.equals("GRB")) {
                    sum++;
                }
            }
        }
        System.out.println(redCount * gNum * bNum - sum);
    }
}